lib/webdriver.py

```diff
@@ -121,35 +127,57 @@ def get_reservations(self, account_monitor: AccountMonitor) -> List[JSON]:
         # instead of requesting again later
         reservations = self._fetch_reservations(driver)
 
-        driver.quit()
+        self._quit_driver(driver)
         return reservations
 
     def _get_driver(self) -> Driver:
-        logger.debug("Starting webdriver for current session")
-        browser_path = self.checkin_scheduler.reservation_monitor.config.browser_path
+        # This environment variable is set in the Docker image
+        is_docker = os.environ.get("AUTO_SOUTHWEST_CHECK_IN_DOCKER") == "1"
 
         driver_version = "mlatest"
-        if os.environ.get("AUTO_SOUTHWEST_CHECK_IN_DOCKER") == "1":
-            # This environment variable is set in the Docker image. Makes sure a new driver
-            # is not downloaded as the Docker image already has the correct driver
+        if is_docker:
+            self._start_display()
+            # Make sure a new driver is not downloaded as the Docker image
+            # already has the correct driver
             driver_version = "keep"
 
+        logger.debug("Starting webdriver for current session")
+        browser_path = self.checkin_scheduler.reservation_monitor.config.browser_path
+
+        # Create a new temporary directory for Chrome profile
+        profile_dir = os.path.join("/tmp", "chrome_profile")
+        os.makedirs(profile_dir, exist_ok=True)
+
+        # Randomly decide whether to use a fresh profile or the persistent one
+        temp_dir = tempfile.mkdtemp() if random.choice([True, False]) else profile_dir
+
         driver = Driver(
             binary_location=browser_path,
             driver_version=driver_version,
-            headless=True,
+            user_data_dir=temp_dir,
+            page_load_strategy="none",
+            headed=is_docker,
+            headless=not is_docker,
             uc_cdp_events=True,
             undetectable=True,
-            is_mobile=True,
+            incognito=True,
         )
         logger.debug("Using browser version: %s", driver.caps["browserVersion"])
 
         driver.add_cdp_listener("Network.requestWillBeSent", self._headers_listener)
+        driver.delete_all_cookies()
 
-        logger.debug("Loading Southwest home page (this may take a moment)")
-        driver.open(BASE_URL)
+        logger.debug("Loading Southwest check-in page (this may take a moment)")
+        driver.set_window_size(random.randint(1024, 1920), random.randint(768, 1080))
+        driver.uc_open_with_reconnect(BASE_URL, 2)
+        driver.wait_for_element("//*[@alt='Check in banner']")
         self._take_debug_screenshot(driver, "after_page_load.png")
-        driver.click("(//div[@data-qa='placement-link'])[2]")
+        time.sleep(random_sleep_duration(1, 2))
+        driver.click("//*[@alt='Check in banner']")
+
+        # Clean up the temporary directory after use
+        shutil.rmtree(driver.user_data_dir, ignore_errors=True)
```

Comment:
Could we wrap this in `with` context statement? I'm concerned the data dir wouldn't get cleaned up if the user interrupts (Ctrl-C) or an exception is thrown before this is executed--happens quite often when the Driver fails to start or hangs for a long time

Reply:
Yes, I had that before but I was getting some 429 on login, I will put that back again hopefully it won't trigger 429 on login - we'll see.

Reply:
> Could we wrap this in `with` context statement? I'm concerned the data dir wouldn't get cleaned up if the user interrupts (Ctrl-C) or an exception is thrown before this is executed--happens quite often when the Driver fails to start or hangs for a long time

it's alright if the data does not get cleaned up if fails, it checks if chrome_profile directory exists and reuses it, its not mandatory to clean it and if it does not take much space.

This setup for some reason does not give @ne0ark errors.